src/core/lib/Modhex.mjs

```diff
@@ -0,0 +1,174 @@
+/**
+ * Modhex functions.
+ *
+ * @author linuxgemini [ilteris@asenkron.com.tr]
+ * @copyright Crown Copyright 2020
+ * @license Apache-2.0
+ */
+
+import Utils from "../Utils.mjs";
+import { toHex, toHexFast, fromHex } from "./Hex.mjs";
+
+/**
+ * Convert a byte array into a modhex string.
+ *
+ * @param {byteArray|Uint8Array|ArrayBuffer} data
+ * @param {string} [delim=" "]
+ * @param {number} [padding=2]
+ * @returns {string}
+ *
+ * @example
+ * // returns "cl bf bu"
+ * toModhex([10,20,30]);
+ *
+ * // returns "cl:bf:bu"
+ * toModhex([10,20,30], ":");
+ */
+export function toModhex(data, delim=" ", padding=2, extraDelim="", lineSize=0) {
+    if (!data) return "";
+    if (data instanceof ArrayBuffer) data = new Uint8Array(data);
+
+    const hexconv = toHex(data, "", padding, "", 0);
+    let modhexconv = "";
+    let output = "";
+
+    for (const letter of hexconv.split("")) {
+        modhexconv += HEX_TO_MODHEX_CONVERSION_MAP[letter];
+    }
+
+    const groupedModhex = modhexconv.match(/.{1,2}/g);
+
+    for (let i = 0; i < groupedModhex.length; i++) {
+        const group = groupedModhex[i];
+        output += group + delim;
+
+        if (extraDelim) {
+            output += extraDelim;
+        }
+        // Add LF after each lineSize amount of bytes but not at the end
+        if ((i !== groupedModhex.length - 1) && ((i + 1) % lineSize === 0)) {
+            output += "\n";
+        }
+    }
+
+    // Remove the extraDelim at the end (if there is one)
+    // and remove the delim at the end, but if it's prepended there's nothing to remove
+    const rTruncLen = extraDelim.length + delim.length;
+    if (rTruncLen) {
+        // If rTruncLen === 0 then output.slice(0,0) will be returned, which is nothing
+        return output.slice(0, -rTruncLen);
+    } else {
+        return output;
+    }
+}
+
+
+/**
+ * Convert a byte array into a modhex string as efficiently as possible with no options.
+ *
+ * @param {byteArray|Uint8Array|ArrayBuffer} data
+ * @returns {string}
+ *
+ * @example
+ * // returns "clbfbu"
+ * toModhexFast([10,20,30]);
+ */
+export function toModhexFast(data) {
+    if (!data) return "";
+    if (data instanceof ArrayBuffer) data = new Uint8Array(data);
+
+    const output = [];
+    const hexconv = toHexFast(data);
+
+    for (const letter of hexconv.split("")) {
+        output.push(HEX_TO_MODHEX_CONVERSION_MAP[letter]);
+    }
+
+    return output.join("");
+}
+
+
+/**
+ * Convert a hex string into a byte array.
+ *
+ * @param {string} data
+ * @param {string} [delim]
+ * @param {number} [byteLen=2]
+ * @returns {byteArray}
+ *
+ * @example
+ * // returns [10,20,30]
+ * fromModhex("cl bf bu");
+ *
+ * // returns [10,20,30]
+ * fromModhex("cl:bf:bu", "Colon");
+ */
+export function fromModhex(data, delim="Auto", byteLen=2) {
+    if (delim !== "None") {
+        const delimRegex = delim === "Auto" ? /[^cbdefghijklnrtuv]/gi : Utils.regexRep(delim);
+        data = data.replace(delimRegex, "");
```

Comment:
Multiline inputs will break if the `delim` is anything other than `Auto` (new line will be converted into `undefined`).

Reply:
With the new test cases, I don't see any issues.